.changeset/techdocs-seahorses-guess.md

```diff
@@ -0,0 +1,28 @@
+---
+'@backstage/plugin-techdocs': patch
+---
+
+Extend the `DocsTable` component to accept custom user-defined options while still providing default values in case of empty.
+
+Example:
+
+```diff
+//...
+
++ const options = {
++   paging: false,
++   search: false,
++   showTitle: false,
++   toolbar: false,
++ };
+
+//...
+
+ <DocsTable
+   entities={entities}
+   columns={columns}
++  options={options}
+   loading={loading}
+   actions={actions}
+ />
```

Comment:
This example may not be clear for someone who is trying to customize the `TechDocsIndexPage` for the first time, could you explain in this example how to create a custom page and use the `DocsTable` there? Or update the example we have in the official documentation and add the link to the documentation in the changeset...

https://github.com/backstage/backstage/blob/master/docs/features/techdocs/how-to-guides.md?plain=1#L107

And I think we can also add the new `options` prop here:
https://github.com/backstage/backstage/blob/master/plugins/techdocs/src/home/components/DefaultTechDocsHome.tsx#L72

What do you think?

Reply:
Example:

**packages/app/src/components/techdocs/CustomTechDocsHome.tsx**

```diff
import React from 'react';
import { Content } from '@backstage/core-components';
import {
  CatalogFilterLayout,
  EntityListProvider,
  EntityOwnerPicker,
  EntityTagPicker,
  UserListPicker,
} from '@backstage/plugin-catalog-react';
import {
  TechDocsPageWrapper,
  TechDocsPicker,
  EntityListDocsTable,
} from '@backstage/plugin-techdocs';

const CustomTechDocsHome = () => {
+ const options = { search: false };
  return (
    <TechDocsPageWrapper>
      <Content>
        <EntityListProvider>
          <CatalogFilterLayout>
            <CatalogFilterLayout.Filters>
              <TechDocsPicker />
              <UserListPicker />
              <EntityOwnerPicker />
              <EntityTagPicker />
            </CatalogFilterLayout.Filters>
            <CatalogFilterLayout.Content>
+             <EntityListDocsTable options={options} />
            </CatalogFilterLayout.Content>
          </CatalogFilterLayout>
        </EntityListProvider>
      </Content>
    </TechDocsPageWrapper>
  );
};

export const techDocsHome = <CustomTechDocsHome />;
```

**packages/app/src/App.tsx**

```diff
import { techDocsHome } from './components/techdocs/CustomTechDocsHome';

..

const routes = (
  <FlatRoutes>
    ...
    <Route path="/docs" element={<TechDocsIndexPage />}>
+    {techDocsHome}
    </Route>
  </FlatRoutes>
);
```

Reply:
thanks fo reviewing, as per suggestion, I have updated the DefaultTechDocsHome component to have the options prop.

I have also added a better example code on the changeset for creating custom page. I didn't edit the how-to-guide since the guide is more about creating a custom techdocs homepage, and less about the DocsTable changes (and since default work just fine with or without the additional support props - like column, actions, or _options_)